pkg/ddevapp/config.go

```diff
@@ -31,6 +33,9 @@ const DDevTLD = "ddev.local"
 // AllowedAppTypes lists the types of site/app that can be used.
 var AllowedAppTypes = []string{"drupal7", "drupal8", "wordpress"}
 
+// Regexp pattern to match any non-alphanumeric characters (except dashes) so they can be replaced.
+var re = regexp.MustCompile(`[^a-zA-Z0-9_-]+`)
```

Comment:
Any reason to not just do a full hostname regex here, since that's our end goal?

It's a lot more wordy, but a full hostname REGEX based on RFC 1123 would be: `^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$` according to https://stackoverflow.com/a/106223